src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js

```diff
@@ -0,0 +1,182 @@
+  attribute = '',
+  identifierRole = 'main',
+  errors,
+}) => (
+  // XXX: we're not using formik's initialErrors yet because it doesn't support arrays of
+  // error messages, which our backend *can* produce.
+  <Formik
+    initialValues={{
+      plugin,
+      attribute,
+      identifierRole,
+    }}
+    onSubmit={(values, actions) => {
+      onSubmit(values);
+      actions.setSubmitting(false);
+    }}
+  >
+    {({handleSubmit}) => (
+      <>
+        <Fieldset>
+          <FormRow>
+            <Field
+              name="plugin"
+              label={
+                <FormattedMessage
+                  description="Variable prefill plugin label"
+                  defaultMessage="Plugin"
+                />
+              }
+              errors={errors.plugin}
+            >
+              <PluginField />
+            </Field>
+          </FormRow>
+
+          <FormRow>
+            <Field
+              name="attribute"
+              label={
+                <FormattedMessage
+                  description="Variable prefill attribute label"
+                  defaultMessage="Attribute"
+                />
+              }
+              errors={errors.attribute}
+            >
+              <AttributeField />
+            </Field>
+          </FormRow>
+
+          <FormRow>
+            <Field
+              name="identifierRole"
+              label={
+                <FormattedMessage
+                  description="Variable prefill identifier role label"
+                  defaultMessage="Identifier role"
+                />
+              }
+              errors={errors.identifierRole}
+            >
+              <IdentifierRoleField />
+            </Field>
+          </FormRow>
+        </Fieldset>
+
+        <SubmitRow>
+          <SubmitAction
+            onClick={event => {
+              event.preventDefault();
+              handleSubmit(event);
+            }}
+          />
+        </SubmitRow>
+      </>
+    )}
+  </Formik>
+);
+
+PrefillConfigurationForm.propTypes = {
+  onSubmit: PropTypes.func.isRequired,
+  plugin: PropTypes.string,
+  attribute: PropTypes.string,
+  identifierRole: PropTypes.string,
+  errors: PropTypes.shape({
+    plugin: PropTypes.arrayOf(PropTypes.string),
+    attribute: PropTypes.arrayOf(PropTypes.string),
+    identifierRole: PropTypes.arrayOf(PropTypes.string),
+  }).isRequired,
+};
+
+const PluginField = () => {
+  const [fieldProps] = useField('plugin');
+  const {setFieldValue} = useFormikContext();
+  const {
+    plugins: {availablePrefillPlugins},
+  } = useContext(FormContext);
+
+  const {value} = fieldProps;
+
+  // reset the attribute value whenever the plugin changes
+  useUpdateEffect(() => {
+    setFieldValue('attribute', '');
+  }, [setFieldValue, value]);
+
+  const choices = availablePrefillPlugins.map(plugin => [plugin.id, plugin.label]);
+  return <Select allowBlank choices={choices} id="id_plugin" {...fieldProps} />;
+};
+
+const AttributeField = () => {
```

Comment:
I think the backend validation is a bit shoddy on this right now. Checked with the master branch, and it has the same problems, except it's even a bit worse because it creates the illusion the first attribute is selected while that's not the actual case!

Leaving it blank at least allows us to show no attribute was selected.

I'll make sure to include this in follow up ticket(s).

Reply:
Check, doing it in a separate ticket is fine by me :+1:

Reply:
Created #4563

Reply:
oh this particular case actually leads to a server error, interesting. Might fix it in this PR after all.

Reply:
working on this in a separate branch, so I'm marking this as resolved so that we can merge this PR